### PR TITLE
Improve MouseDispatcher perf

### PR DESCRIPTION
**Pull request by themadcreator**
- Use MouseEvent.offsetX/Y instead of TransformAwareTranslator
- Attach listener to component root instead of document

### CODE REVIEW

**Comment by themadcreator** [comment]
In a page with many charts, we see a major performance improvement over previous approach.

### Old & Busted
tat-bad

### New Hotness
image

File: src/dispatchers/mouseDispatcher.ts [hunk 1]
```diff
@@ -191,11 +188,11 @@ export class Mouse extends Dispatcher {
       throw new Error("Invalid scope '" + scope + "', must be 'element' or 'page'");
     }
     if (scope === "page" || this.eventInside(component, event)) {
-      const newMousePosition = this._translator.computePosition(event.clientX, event.clientY);
-      if (newMousePosition != null) {
-        this._lastMousePosition = newMousePosition;
-        this._callCallbacksForEvent(eventName, this.lastMousePosition(), event);
-      }
+      const origin = component.originToRoot();
```
Comment (hellochar) [inline src/dispatchers/mouseDispatcher.ts:191]: should we just delete the Translator? Is it used elsewhere?

**Review by hellochar — APPROVED** [review]

File: src/dispatchers/mouseDispatcher.ts (hunk 1 above)
Comment (themadcreator) [reply src/dispatchers/mouseDispatcher.ts:191]: Yeah, i'll delete

Comment (themadcreator) [reply src/dispatchers/mouseDispatcher.ts:191]: can't delete because it's used for touch events. see follow up PR for perf update there.

**Pull request merged**